nodes/analyzer/weighted_vector_sum.py

```diff
@@ -0,0 +1,453 @@
+            if len(input_density_s_2)==1 and len(input_vertices_s_3)>1:
+                input_density_s_2 = [[d] for d in input_density_s_2[0]]
+
+            # if lists of edges or faces is less vertices then append at the end of edges and faces empty list to extend it later with empty items.
+            # verts=[[v1,v2,v3],[v4,v5,v6]]
+            # edges=[[[0,1],[1,2]]]=>[[[0,1], [1,2]],[]]
+            # faces=[[[0,1,2]]]=>[[[0,1,2]]],[]]
+            # this allow skip unmanifold centers
+            if len(input_edges_s_3)<len(input_vertices_s_3):
+                input_edges_s_3.append([[]]) # like default
+            if len(input_polygons_s_3)<len(input_vertices_s_3):
+                input_polygons_s_3.append([[]]) # like default
+
+            _, _input_edges_s_3, _input_polygons_s_3, _input_mass_vert_s_2, _input_density_s_2 = match_long_repeat( [input_vertices_s_3, input_edges_s_3[:len(input_vertices_s_3)], input_polygons_s_3[:len(input_vertices_s_3)], input_mass_vert_s_2[:len(input_vertices_s_3)], input_density_s_2[:len(input_vertices_s_3)] ])
+            meshes = match_long_repeat([input_vertices_s_3, _input_edges_s_3, _input_polygons_s_3, _input_mass_vert_s_2, _input_density_s_2])
+
+            for vertices_I, edges_I, faces_I, mass_of_vertices_I, density_I in zip(*meshes):
+                faces_I_np = np.array(faces_I)
+                if faces_I_np.size==0:
+                    result_mask.append(False)
+                else:
+                    result_mask.append(True)
+
+                    # triangulate mesh for 'FACES' mode
+                    bm_I = bmesh_from_pydata(vertices_I, edges_I, faces_I, markup_face_data=True, normal_update=True)
+                    b_faces = []
+                    for face in bm_I.faces:
+                        b_faces.append(face)
+                    res = bmesh.ops.triangulate(
+                        bm_I, faces=b_faces, quad_method=self.quad_mode, ngon_method=self.ngon_mode
+                    )
+                    new_vertices_I, new_edges_I, new_faces_I = pydata_from_bmesh(bm_I)
+                    bm_I.free()
+
+                    result_vertices.append(new_vertices_I)
+                    result_edges.append(new_edges_I)
+                    result_polygons.append(new_faces_I)
+                    
+                    verts_I            = np.array(new_vertices_I)
+                    faces_I            = np.array(new_faces_I)
+                    tris_I             = verts_I[faces_I]
+                    areases_I          = np.linalg.norm(np.cross(tris_I[:,1]-tris_I[:,0], tris_I[:,2]-tris_I[:,0]) / 2.0, axis=1)
+                    area_I             = areases_I.sum()
+                    tris_centers_I     = tris_I.sum(axis=1) / 3.0
+                    center_mass_mesh_I = (tris_centers_I * areases_I[np.newaxis].T).sum(axis=0) / area_I
+                    mass_I             = area_I * density_I[0]
```

Comment:
I'd say core algorithms should be moved to sverchok.utils.geom, and just be called from the node.